android-reactive-location/src/main/java/pl/charmas/android/reactivelocation/observables/BaseObservable.java

```diff
@@ -30,7 +30,20 @@ protected BaseObservable(Context ctx, Api<? extends Api.ApiOptions.NotRequiredOp
     @Override
     public void call(Subscriber<? super T> subscriber) {
 
-        final GoogleApiClient apiClient = createApiClient(subscriber);
+        final ApiClientConnectionCallbacks apiClientConnectionCallbacks = new ApiClientConnectionCallbacks(subscriber);
+
+        GoogleApiClient.Builder apiClientBuilder = new GoogleApiClient.Builder(ctx);
+
+        for (Api<? extends Api.ApiOptions.NotRequiredOptions> service : services) {
+            apiClientBuilder.addApi(service);
+        }
+
+        apiClientBuilder.addConnectionCallbacks(apiClientConnectionCallbacks);
+        apiClientBuilder.addOnConnectionFailedListener(apiClientConnectionCallbacks);
+
+        final GoogleApiClient apiClient = apiClientBuilder.build();
+        apiClientConnectionCallbacks.setClient(apiClient);
+
```

Comment:
It is different. This setup allow me to keep a reference to the  `apiClientConnectionCallbacks` and unregister the callbacks when unsubscribed. 
``` 
subscriber.add(Subscriptions.create(new Action0() {
            @Override
            public void call() {
                if (apiClient.isConnected() || apiClient.isConnecting()) {
                    onUnsubscribed(apiClient);
                    apiClient.disconnect();
                    apiClient.unregisterConnectionFailedListener(apiClientConnectionCallbacks);
                    apiClient.unregisterConnectionCallbacks(apiClientConnectionCallbacks);
                }
            }
        }));